### DISCUSSION TITLE
button not disabling

### DISCUSSION THREAD

**Discussion by ToBinio** [nuxt/nuxt#25308] [Questions] [answered]
it seems like the `:disabled` will not be changed when hydrating

I have this example code:

```vue
<script setup lang="ts">
let state = ref(false);

if (process.client) {
  // some code which should only be called on the client e.g localStorage
  state.value = true;
}

</script>

<template>
  {{ state }}
  <button :disabled="state">should be disabled</button>
</template>
```

![grafik](https://github.com/nuxt/nuxt/assets/81473300/5e0d7f35-489d-4f85-ab14-e818268e6e9c)

in this example the state text next to the button is updated but the `:disabled` of the button isn't

I am obviously also getting a hydration warning

how should i write this code so it works?
thanks in advanced

**Comment by TheAlexLichter** [comment, marked as answer]
You are forcing a hydration error this way. Use `onMounted` or `onBeforeMount` for "client-side-only" code.

**Reply by ToBinio** [reply]
`onMounted` worked

thanks a lot for the quick answer and thanks for the YouTube videos too

**Reply by TheAlexLichter** [reply]
You are welcome! 🙏
